src/client/components/ConnectedIntlProvider/index.js

```diff
@@ -3,28 +3,30 @@ import { connect } from 'react-redux';
 import { IntlProvider } from 'react-intl';
 import { onlyUpdateForKeys } from 'recompose';
 import PropTypes from 'prop-types';
+import AppError from '../AppError';
 
-const ConnectedIntlProvider = ({ children, locale, messages }) => {
-  if (!locale) return null;
-  return (
-    <IntlProvider
-      locale={locale}
-      messages={messages}
-    >
-      {children}
-    </IntlProvider>
-  );
+const ConnectedIntlProvider = ({ children, locale, message, messages = {} }) => {
+  if (locale) {
+    return (
+      <IntlProvider locale={locale} messages={messages}>
+        {children}
+      </IntlProvider>
+    );
+  }
+  return <AppError isErrorFound={message.header ? true : false} />; //eslint-disable-line 
```

Comment:
We must not return an AppError, because yet we do not have an error, just we do not have the locale.
* rename AppError as SplashApp
* do not inject message in SplashApp
* connect SplashApp like App (but only for  message prop)
* if message exists, render an error message, else just a message about page loading ... have a look at http://blueprintjs.com/docs/#core/components/progress/skeleton